examples/check_scan_jobs_status.py

```diff
@@ -0,0 +1,224 @@
+        else:
+            return scan_history[-1:]
+
+    def _get_most_recent_jobs(self, job_types, jobs):
+        most_recent_l = list()
+        for job_type in job_types:
+            filtered_to_type = list(filter(lambda j: j['jobSpec']['jobType'] == job_type, jobs))
+            most_recent = sorted(filtered_to_type, key = lambda j: j['startedAt'])[-1]
+            most_recent_l.append(most_recent)
+        return most_recent_l
+
+    def _get_related_jobs(self, most_recent_scans, project_name, version_name):
+        later_than = min([s['createdAt'] for s in most_recent_scans])
+        later_than = timestring.Date(later_than).date
+        pv_name = "{} {}".format(project_name, version_name)
+        scan_loc_name = most_recent_scans[0]['scan_details']['name']
+        scan_ids = [mrs['scan_details']['scanSourceId'] for mrs in most_recent_scans]
+
+        jobs = CodeLocationStatusChecker.jobs
+
+        jobs_with_job_descriptions = list(filter(lambda j: 'jobEntityDescription' in j, jobs))
+
+        pv_jobs = list(filter(
+            lambda j: pv_name in j.get('jobEntityDescription', []) and timestring.Date(j['createdAt']).date > later_than, 
+            jobs_with_job_descriptions))
+        pv_job_types = set([pj['jobSpec']['jobType'] for pj in pv_jobs])
+
+        s_jobs = list(filter(
+            lambda j: scan_loc_name in j.get('jobEntityDescription', []) or j['jobSpec']['entityKey']['entityId'] in scan_ids,
+            jobs))
+        # TODO: Filter again for later_than? or pre-filter for later_than?
+        s_job_types = set([sj['jobSpec']['jobType'] for sj in s_jobs])
+
+        most_recent_pv_jobs = self._get_most_recent_jobs(pv_job_types, pv_jobs)
+        most_recent_s_jobs = self._get_most_recent_jobs(s_job_types, s_jobs)
+
+        combined_recent_jobs = most_recent_pv_jobs + most_recent_s_jobs
+        # de-dup; see https://stackoverflow.com/questions/11092511/python-list-of-unique-dictionaries
+        combined_recent_jobs = list({j['id']: j for j in combined_recent_jobs}.values())
+
+        return combined_recent_jobs
+
+    def has_snippet_scan(self, scan_history):
+        return any([s['scan_details']['scanType'].lower() == "snippet" for s in scan_history[:2]])
+
+    def succeeded(self):
+        if self.has_snippet_scan(self.scan_history):
+            scans_to_check = self.scan_history[:2]
+        else:
+            scans_to_check = self.scan_history[-1:]
+
+        assert len(scans_to_check) > 0, "Uh oh, no scans to check. That shouldn't happen."
+        assert len(self.related_jobs) > 0, "Uh oh, no related jobs to check. That shouldn't happen."
+
+        checks = [s['status'].lower() == "complete" for s in scans_to_check] 
+        checks.extend([j['status'].lower() == "completed" for j in self.related_jobs])
+
+        self.failed_jobs = [j for j in self.related_jobs if j['status'].lower() != "completed"]
+        self.failed_scans = [s for s in scans_to_check if s['status'].lower() != "complete"]
```

Comment:
are in-progress jobs considered failed?

Reply:
Short answer: no. However, from the POV of this code we are checking for success and anything that is not success is a fail. 

I think it makes sense though to extend the code to differentiate between in-progress and outright fail. I'm worried about the case of a scan/job that is running and will not finish. We've had cases where scans hang and I want to make sure that situation is a 'fail' in the sense that a non-zero exit status is set if/when all jobs and scans have not complete(d). Make sense?

Reply:
makes sense